src/sage/combinat/k_regular_sequence.py

```diff
@@ -240,6 +336,227 @@ def __iter__(self):
+    @minimize_result
+    def regenerated(self):
+        r"""
+        Return a `k`-regular sequence that satisfies
+        `\mu[0] \mathit{right} = \mathit{right}`.
+
+        INPUT:
+
+        - ``minimize`` -- (default: ``None``) a boolean or ``None``.
+          If ``True``, then :meth:`~RecognizableSeries.minimized` is called after the operation,
+          if ``False``, then not. If this argument is ``None``, then
+          the default specified by the parent's ``minimize_results`` is used.
+
+        OUTPUT:
+
+        A :class:`kRegularSequence`
+
+        EXAMPLES::
+
+            sage: Seq2 = kRegularSequenceSpace(2, ZZ)
+
+        The following linear representation of `S` is chosen bad (is
+        degenerated, see :meth:`is_degenerated`), as `\mu(0)` applied on
+        `\mathit{right}` does not equal `\mathit{right}`::
+
+            sage: S = Seq2((Matrix([2]), Matrix([3])), vector([1]), vector([1]),
+            ....:          allow_degenerated_sequence=True)
+            sage: S
+            2-regular sequence 1, 3, 6, 9, 12, 18, 18, 27, 24, 36, ...
+            sage: S.is_degenerated()
+            True
+
+        However, we can regenerate the sequence `S`::
+
+            sage: H = S.regenerated()
+            sage: H
+            2-regular sequence 1, 3, 6, 9, 12, 18, 18, 27, 24, 36, ...
+            sage: H.linear_representation()
+            ((1, 0),
+             Finite family {0: [ 0  1]
+                               [-2  3],
+                            1: [3 0]
+                               [6 0]},
+             (1, 1))
+            sage: H.is_degenerated()
+            False
+
+        TESTS::
+
+            sage: S = Seq2((Matrix([2]), Matrix([3])), vector([1]), vector([1]),
+            ....:          allow_degenerated_sequence=True)
+            sage: H = S.regenerated(minimize=False)
+            sage: H.linear_representation()
+            ((1, 1),
+             Finite family {0: [1|0]
+                               [-+-]
+                               [0|2],
+                            1: [0|0]
+                               [-+-]
+                               [3|3]},
+             (1, 0))
+            sage: H.is_degenerated()
+            False
+        """
+        if not self.is_degenerated():
+            return self
+
+        from sage.matrix.constructor import Matrix
+        from sage.matrix.special import zero_matrix, identity_matrix
+        from sage.modules.free_module_element import vector
+
+        P = self.parent()
+        dim = self.dimension()
+        Z = zero_matrix(dim)
+        I = identity_matrix(dim)
+
+        itA = iter(P.alphabet())
+        z = next(itA)
+        mu = {z: Matrix.block([[I, Z], [Z, self.mu[z]]])}
+        mu.update((r, Matrix.block([[Z, Z], [self.mu[r], self.mu[r]]]))
+                  for r in itA)
+
+        return P.element_class(
+            P, mu,
+            vector(2*tuple(self.left)),
+            vector(tuple(self.right) + dim*(0,)))
+
```

Comment:
This version here constructs a `2*dim`-dimensional linear representation. The corresponding right vector valued sequence $\tilde v(n)$ is $(v(n)[n=0], v(n)[n\neq 0])^\top$ where $v(n)$ is the right vector valued sequence of the original linear representation.

Theorem B of [Heuberger, C., Krenn, D. & Lipnik, G.F. Asymptotic Analysis of q-Recursive Sequences. Algorithmica 84, 2480–2532 (2022)](https://doi.org/10.1007/s00453-022-00950-y) constructs a $dim+1$-dimensional linear representation at no significantly higher cost.

Therefore, I suggest to 
- replace the implementation here by the one of the above mentioned paper;
- write a brief Algorithm section for the docstring which refers to that paper.

Reply:
Indeed, this should be changed. I'll prepare a patch...

Reply:
d665cd6d31e